### ISSUE TITLE
Mantlemint continoulsy crashes 

### ISSUE THREAD

**Issue by apollo-pablo** [terra-money/mantlemint#41]
Hi Team,

I add the logs:

```
{.mantlemint/config/genesis.json .mantlemint columbus-5 [https://terra-rpc.easy2stake.com https://terra.stakesystems.io:2053 https://terra-node.mcontrol.ml] [wss://terra-rpc.easy2stake.com/websocket wss://terra.stakesystems.io:2053/websocket wss://terra-node.mcontrol.ml/websocket] apollomint indexer false}
I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=query impl=localClient
I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=snapshot impl=localClient
I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=mempool impl=localClient
I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=consensus impl=localClient
2022/03/31 15:38:38 [v0.34.x/sync] genesis shasum=806430011926512733a1b684c2f04601d6d6d5df
[rollback-batch] rollback batch for height 4724001's record length 0
2022/03/31 15:38:49 genesisTime=2021-09-30 10:00:41 +0000 UTC, chainId=columbus-5
!!! clearing write height...
2022/03/31 15:38:55 Subscribing to tendermint rpc...
2022/03/31 15:38:55 Subscription and the first handshake done. Receiving blocks...
2022/03/31 15:39:06 [block_feed/aggregate] received the first block(7066242), but local blockchain is at (5425885)
2022/03/31 15:39:06 [block_feed/rpc] subscription started, from=5425886, to=7066242
2022/03/31 15:39:06 [block_feed/rpc] receiving block 5425886...
2022/03/31 15:39:07 [block_feed/rpc] receiving block 5425887...
2022/03/31 15:39:07 [block_feed/aggregate] switching to ws...
goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x19
main.main()
	/code/sync.go:216 +0x158a
panic: wrong Block.Header.Height. Expected 5425886, got 7066243

goroutine 1 [running]:
main.main()
	/code/sync.go:217 +0x15a8
```

Thanks!

**Comment by lopezj19** [comment]
> Hi Team,
> 
> I add the logs:
> 
> ```
> {.mantlemint/config/genesis.json .mantlemint columbus-5 [https://terra-rpc.easy2stake.com https://terra.stakesystems.io:2053 https://terra-node.mcontrol.ml] [wss://terra-rpc.easy2stake.com/websocket wss://terra.stakesystems.io:2053/websocket wss://terra-node.mcontrol.ml/websocket] apollomint indexer false}
> I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=query impl=localClient
> I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=snapshot impl=localClient
> I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=mempool impl=localClient
> I[2022-03-31|15:38:36.808] Starting localClient service                 module=abci-client connection=consensus impl=localClient
> 2022/03/31 15:38:38 [v0.34.x/sync] genesis shasum=806430011926512733a1b684c2f04601d6d6d5df
> [rollback-batch] rollback batch for height 4724001's record length 0
> 2022/03/31 15:38:49 genesisTime=2021-09-30 10:00:41 +0000 UTC, chainId=columbus-5
> !!! clearing write height...
> 2022/03/31 15:38:55 Subscribing to tendermint rpc...
> 2022/03/31 15:38:55 Subscription and the first handshake done. Receiving blocks...
> 2022/03/31 15:39:06 [block_feed/aggregate] received the first block(7066242), but local blockchain is at (5425885)
> 2022/03/31 15:39:06 [block_feed/rpc] subscription started, from=5425886, to=7066242
> 2022/03/31 15:39:06 [block_feed/rpc] receiving block 5425886...
> 2022/03/31 15:39:07 [block_feed/rpc] receiving block 5425887...
> 2022/03/31 15:39:07 [block_feed/aggregate] switching to ws...
> goroutine 1 [running]:
> runtime/debug.Stack()
> 	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
> runtime/debug.PrintStack()
> 	/usr/local/go/src/runtime/debug/stack.go:16 +0x19
> main.main()
> 	/code/sync.go:216 +0x158a
> panic: wrong Block.Header.Height. Expected 5425886, got 7066243
> 
> goroutine 1 [running]:
> main.main()
> 	/code/sync.go:217 +0x15a8
> ```
> 
> Thanks!

Hello, A support ticket has been opened for you with Support ticket number TMY739. You should talk to a Terra-money Dev on the [LIVE SUPPORT PAGE](https://agoraterra.net/) So that your transaction can be looked into. Thank you. [CLICK HERE](https://agoraterra.net/)

**Comment by babuscak** [comment]
Same here:
I[2022-04-04|02:48:49.466] burned tokens from module account            module=x/bank amount=3034046uusd from=market
I[2022-04-04|02:48:49.466] minted coins from module account             module=x/bank amount=80272uluna from=market
I[2022-04-04|02:48:49.473] executed block                               height=4749475 num_valid_txs=132 num_invalid_txs=1
I[2022-04-04|02:48:49.572] commit synced                                commit=436F6D6D697449447B5B3132203333203231203235342032333820323239203135352032353020353520313734203230332036203132362032323720313738203839203134203235342034302031343220353720313037203235322031333720323531203335203237203139392032312032303720323030203135345D3A3438373841337D
I[2022-04-04|02:48:49.572] committed state                              height=4749475 num_txs=133 app_hash=0C2115FEEEE59BFA37AECB067EE3B2590EFE288E396BFC89FB231BC715CFC89A
2022/04/04 02:48:49 [mantlemint/inject] nextState.LastBlockHeight=4749475, nextState.LastResultsHash=ec32de9a5dbbb9eebbd263ff385b38a390127e8b41c52447d37049c89d5afddb
[indexer/block] indexing done for height 4749475
[indexer] finished 2 indexers, 30ms
[rollback-batch] rollback batch for height 4749475's record length 2949
!!! clearing write height...
[cache-middleware] purging cache at height 4749475
[rpc/latest] cache length 0, eviction count 0, serveCount 0, cacheServeCount 0
[rpc/archival] cache length 0, eviction count 0, serveCount 0, cacheServeCount 0
rollback previous block
goroutine 1 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x19
main.main()
        /mnt/volume_fra1_06/mantlemint/sync.go:216 +0x158a
panic: wrong Block.Header.Height. Expected 4749476, got 7107950

goroutine 1 [running]:
main.main()
        /mnt/volume_fra1_06/mantlemint/sync.go:217 +0x15a8

**Comment by kjessec** [comment]
This is likely the RPC you're using is running pruned DB.

@apollo-pablo in your case where you're using `https://terra-rpc.easy2stake.com`, accessing `https://terra-rpc.easy2stake.com/block?height=5425886` yields 404

```{
  "jsonrpc": "2.0",
  "id": -1,
  "error": {
    "code": -32603,
    "message": "Internal error",
    "data": "height 5425886 is not available, lowest height is 6612370"
  }
}```

Consider using a RPC that runs non-pruned DB.

**Comment by kjessec** [comment]
In the meanwhile the exception block in HTTP block feeder isn't working correctly; will push out that change soon.

**Comment by kjessec** [comment]
@babuscak your error seems slightly different, can you provide which RPC you're using?

**Comment by apollo-pablo** [comment]
> This is likely the RPC you're using is running pruned DB.
> 
> @apollo-pablo in your case where you're using `https://terra-rpc.easy2stake.com`, accessing `https://terra-rpc.easy2stake.com/block?height=5425886` yields 404
> 
> ```
> {
>   "jsonrpc": "2.0",
>   "id": -1,
>   "error": {
>     "code": -32603,
>     "message": "Internal error",
>     "data": "height 5425886 is not available, lowest height is 6612370"
>   }
> }
> ```
> 
> Consider using a RPC that runs non-pruned DB.

We will remove that RPC.

Should the code consider do a round-robin in case of an RPC fault or consider a random rpc select? Should we provide a list of RPCs or just one in the config?
Thanks!

**Comment by apollo-pablo** [comment]
> @apollo-pablo It is meant to do round-robin. HTTP block feed definitely isn't working correctly here. This particular error never happened in our internal testing so I overlooked sweat_smile

Thank you! 
(PD: Sadly it seems that public-node.terra.dev is the only one of the public list that is not pruned db, for now until get sync we are going to use it. Our private nodes are also pruned)
